File: fapi/src/main/java/by/kapitonov/cinema/fapi/service/FilmSessionService.java
package by.kapitonov.cinema.fapi.service;

import by.kapitonov.cinema.fapi.model.FilmSession;
import by.kapitonov.cinema.fapi.rest.response.ApiResponse;
import by.kapitonov.cinema.fapi.rest.response.PageResponse;
import by.kapitonov.cinema.fapi.service.dto.CreateFilmSessionDTO;

public interface FilmSessionService {

    PageResponse<FilmSession> getAll(int page, int size);
    PageResponse<FilmSession> getAllActiveSessionByHallId(Long hallId, int page, int size);

    FilmSession getOne(Long sessionId);

    ApiResponse create(CreateFilmSessionDTO filmSessionDTO);

}

File: frontend/src/app/service/film-session.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {Page} from "../model/Page";
import {FilmSession} from "../model/FilmSession";
import { CreateFilmSessionDTO } from '../model/dto/film-session/CreaetFilmSessionDTO';
import { ApiResponse } from '../model/ApiResponse';

@Injectable({
  providedIn: 'root'
})
export class FilmSessionService {

  private url: string = `http://localhost:8091/api/film-sessions`;

  constructor(private http: HttpClient) { }

  public getAllActive(hallId: number, page: number, size: number): Observable<Page<FilmSession>> {
    return this.http.get<Page<FilmSession>>(`${this.url}/${hallId}/active?page=${page}&size=${size}`);
  }

  public getAll(hallId: number, page: number, size: number): Observable<Page<FilmSession>> {
    return this.http.get<Page<FilmSession>>(`${this.url}/${hallId}/all?page=${page}&size=${size}`);
  }

  public getOne(sessionId: number): Observable<FilmSession> {
    return this.http.get<FilmSession>(`${this.url}/${sessionId}`);
  }

  public create(filmSessionDTO: CreateFilmSessionDTO): Observable<ApiResponse> {
    return this.http.post<ApiResponse>(`${this.url}`, filmSessionDTO);
  }

  public changeStatus(filmSessionId: number, statusName: string): Observable<ApiResponse> {
    return this.http.put<ApiResponse>(`${this.url}/${filmSessionId}/status?status=${statusName}`, null);
  }
}
